test/factories/serial.rb

```diff
@@ -0,0 +1,17 @@
+FactoryGirl.define do
+  factory :serial do
+    sequence(:title) {|n| "Serial #{n}" }
+    sequence(:slug) {|n| "serial#{n}" }
+    description { "#{title} description\n" }
+
+    trait :whatever do
+      site
+    end
+
+    trait :with_posts do
+      after :create do |serial, _evaluator|
+        serial.posts << build(:post, :whatever, site: serial.site)
```

Comment:
I know `Post#category_id` is mandatory.
`:post`'s `:whatever` trait creates category automatically.
So I did not add `category` here.